p2p/server/msgserver.go

```diff
@@ -103,9 +103,11 @@ func NewMsgServer(ctx context.Context, network Service, name string, requestLife
 
 // Close stops the MessageServer
 func (p *MessageServer) Close() {
-	p.exit <- struct{}{}
-	<-p.exit
+	p.With().Info("closing MessageServer")
+	close(p.exit)
```

Comment:
i think the point was to wait until readLoop exits. it is started without workerCount.Add(1). otherwise it is possible that that Add(1) in readLoop will be called after Wait here

Reply:
great catch!!! thanks for the review. much appreciated.